app/src/main/java/com/dessalines/thumbkey/utils/Types.kt

```diff
@@ -57,9 +77,34 @@ data class KeyC(
             else -> null
         },
     val capsModeDisplay: KeyDisplay? = null,
-    val color: ColorVariant = ColorVariant.SECONDARY,
     val size: FontSizeVariant = FontSizeVariant.SMALL,
-)
+    val color: ColorVariant =
+        when (size) {
+            FontSizeVariant.LARGE -> ColorVariant.PRIMARY
+            else -> ColorVariant.SECONDARY
+        },
+) {
+    constructor(
```

Comment:
I'm not a big fan of constructors for data classes, especially since all it does is obfuscate the fields.

Reply:
In this PR, I was mostly thinking of the ergonomics of writing and reading keyboard definitions. From this point of view, it doesn't really matter what the actual fields of a data class are. What does matter, however, is how "noisy" the most common use cases are. And `KeyC("X", ...)` just reads better than `KeyC(CommitText("X"), ...)`, in my opinion.

There are a couple of other ways I could achieve a similar effect. For example, having both `text` and `action` fields on a `KeyC`, and having one override the other. Or making this overload a standalone function or a static method, like `textKey("X", ...)` or `KeyC.text("X", ...)`. I like the constructor overload more than those, but the final say is on you.

Reply:
I spose I'm good with the constructor overload then, especially since compile-time-checked anyway.

My main concern is that if we add too many different ways of creating keys / keyboards, then we're going to get a lot of conflicts when we have to do refactors. But I think you've done it in a way that's justified here.

Thx for all this work, I'm wondering if you used vim macros or something to be able to do this for all the keyboards. How did you do it?

Reply:
> My main concern is that if we add too many different ways of creating keys / keyboards, then we're going to get a lot of conflicts when we have to do refactors. But I think you've done it in a way that's justified here.

Yeah, I see where you're coming from. My original draft had actually included more overloads - for example, to allow for an icon not to be wrapped in an `IconDisplay`. But I removed them, for much the same reasons you describe.

> I'm wondering if you used vim macros or something to be able to do this for all the keyboards. How did you do it?

I was mostly using the regular "find and replace in files" feature of VS Code, which is my main IDE. It accepts multi-line regular expressions to search for, and allows to use capture groups in substitutions. AFAIK, it uses `ripgrep` to implement this feature.

I also edited stuff by hand, of course.

Reply:
Nice